src/Pages/ProjectDetails.jsx

```diff
@@ -1,3 +1,63 @@
+import { useEffect, useState } from "react";
+import { useHistory, useParams } from "react-router";
+
 export default function ProjectDetails() {
-  return;
+  const history = useHistory();
+  const { id, title } = useParams();
+  const tragetId = +id;
+  const [project, setProject] = useState(null);
+
+  useEffect(() => {
+    const url = `${process.env.REACT_APP_API_URL}/projects/${tragetId}`;
+    fetch(url)
+      .then((res) => res.json())
+      .then((projectData) => {
+        setProject(projectData);
+      });
+  }, [tragetId]);
+
+  if (project === null) {
+    return "loading";
+  }
+
+  if (project.title !== title) {
+    history.push(`/projects/${id}/${project.title}`);
+  }
+
+  const { title: projectTitle, description, goal, user, donations } = project;
+  const { name, profile } = user;
+  const { country } = profile;
+
+  const countDonations = (donations) => {
+    let result = 0;
+    donations.forEach((donation) => (result += donation));
+    return result;
```

Comment:
You could make this function a one-liner by using `Array.prototype.reduce`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

```
return donations.reduce((a, b) => a + b, 0);
```